graphql/provider.go

```diff
@@ -40,10 +62,66 @@ func graphqlConfigure(ctx context.Context, d *schema.ResourceData) (interface{},
 		GQLServerUrl:   d.Get("url").(string),
 		RequestHeaders: d.Get("headers").(map[string]interface{}),
 	}
+
+	if oauth2LoginQueryValueAttribute := d.Get("oauth2_login_query_value_attribute"); d.Get("oauth2_login_query") != "" && oauth2LoginQueryValueAttribute != "" {
```

Comment:
I'm thinking we need an `else` on this that returns some validation diags to the user letting them know that if they set the oauth2_login_query_value_attribute than they must also have the query and oauth2_login_query_variables set

Reply:
Made it an error if some but not all OAuth 2.0 attributes are set:

```
╷
│ Error: Incomplete OAuth 2.0 provider configuration
│
│   with provider["registry.terraform.io/sullivtr/graphql"],
│   on providers.tf line 1, in provider "graphql":
│    1: provider "graphql" {
│
│ All three attributes must be set: `oauth2_login_query`, `oauth2_login_query_variables` and `oauth2_login_query_value_attribute`.
╵
```